### DISCUSSION TITLE
Error if inputs are changed after a task runs

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [vercel/turborepo#8710] [Ideas]
### Goals

1. I want to aggressively know when cache would be invalidated
2. I don't want unexpected changes to happen
3. I want a tool to help me define the turbo.json files

### Background

Often times, when running task, outputs may overlap with inputs.
Ideally these are totally separated.

So if a file detected as an output is _also_ an input, I would like to have turbo error

### Proposal

error on overlap of output and input files after a task run

**Comment by anthonyshew** [comment]
This makes sense to me. Would you expect this to be an opt-in configuration or the default behavior?

**Reply by NullVoxPopuli** [reply]
I would like it to be default, but I could live with the argument that that would be a breaking change (to be made default at the next major), even though the accidental dirtying behavior _should_ never be desired.
